### PR TITLE
Fallback to rsync from https when trust anchor certificate can not be loaded

### PR DESCRIPTION
**Pull request by ties**
Quick and slightly dirty implementation of fallback behaviour. Would like it if the implementation is slightly cleaner.

### CODE REVIEW

File: rpki-validator/src/main/java/net/ripe/rpki/validator3/storage/data/TrustAnchor.java [hunk 1]
```diff
@@ -98,4 +101,11 @@ public X509ResourceCertificate getCertificate() {
     public void markInitialCertificateTreeValidationRunCompleted() {
         this.initialCertificateTreeValidationRunCompleted = true;
     }
+
+    public List<URI> getLocationsByPreference() {
+        return locations.stream()
+            .sorted(Comparator.naturalOrder())
```
Comment (wibisono) [inline rpki-validator/src/main/java/net/ripe/rpki/validator3/storage/data/TrustAnchor.java:107]: Just to confirm the intent, sorting in the ends put https in front of rysnc, right?

**Comment by ties** [comment]
Yup, dat is the intent.

Perhaps I should document that instead of it being implicit.

File: rpki-validator/src/main/java/net/ripe/rpki/validator3/storage/data/TrustAnchor.java (hunk 1 above)
Comment (ties) [reply rpki-validator/src/main/java/net/ripe/rpki/validator3/storage/data/TrustAnchor.java]: Sorting would re-order urls with the same scheme alphabetically. I changed this.

File: rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/validation/TrustAnchorValidationService.java [hunk 2]
```diff
@@ -121,41 +123,41 @@ public void validate(long trustAnchorId) {
             return new TrustAnchorValidationRun(trustAnchorRef, trustAnchor.getLocations().get(0));
         });
 
+        final ValidationLocation trustAnchorValidationLocation = new ValidationLocation(validationRun.getTrustAnchorCertificateURI());
+        ValidationResult validationResult = ValidationResult.withLocation(trustAnchorValidationLocation);
+
         boolean updatedTrustAnchor = false;
+        final List<URI> rankedTrustAnchorLocations = trustAnchor.getLocationsByPreference();
         try {
+            boolean loadedAtLeastOneTACertificate = false;
```
Comment (lolepezy) [inline rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/validation/TrustAnchorValidationService.java:132]: Can you hide this logic into a separate method? Here we basically only want to know if TA cert was fetched or not, without all the details.

**Pull request merged**